__tests__/service/onesignal/OneSignalUtil.test.ts

```diff
@@ -33,7 +34,7 @@ describe('TESTING OneSignalUtil', () => {
 
       // Assertion
       expect(OneSignal.User.removeTag).toHaveBeenCalledTimes(3);
-      expect(OneSignal.User.addTag).toHaveBeenCalledTimes(2);
+      expect(OneSignal.User.addTag).toHaveBeenCalledTimes(3);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change in the expected number of times `OneSignal.User.addTag` is called from 2 to 3 is consistent with the addition of the `hasUserPosted` field. If `hasUserPosted` is true, an additional call to `OneSignal.User.addTag` is made, hence the increase in the expected count.

```diff
-      expect(OneSignal.User.addTag).toHaveBeenCalledTimes(2);
+      expect(OneSignal.User.addTag).toHaveBeenCalledTimes(3);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->